modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -207,7 +222,7 @@ def get_index_and_columns(df):
     return len(df.index), len(df.columns)
 
 
-@ray.remote(num_returns=4)
+@ray.remote(num_returns=4 if EnablePartitionIPs.get() else 3)
```

Comment:
Ideally this should be changed so that `deploy_ray_func()` is regenerated each time `EnablePartitionsIPs` is changed.
I suggest rewriting it like this (untested):
```python
def _deploy_ray_func(call_queue, parition):
    # actual code of the func
    return result

deploy_ray_func = None

def _update_deploy(enabled):
    global deploy_ray_func
    deploy_ray_func = ray.remote(num_returns=4 if enabled else 3)(_deploy_ray_func)
EnablePartitionIPs.subscribe(_update_deploy)
```

Reply:
I am not sure it can be fixed by following your suggestion. Fixed by another way.

Reply:
I don't see it fixed in current state of PR. Also my proposal should be valid :)